### ISSUE TITLE
Add in the transition to another State Machine the required condition bool value to be activated in the AnimationTree.

### ISSUE THREAD

**Issue by Sfoyarbide** [godotengine/godot-proposals#8256]
### Describe the project you are working on

Working on the animations of my JRPG game.

### Describe the problem or limitation you are having in your project

I have realized that there's no way to set the transition's condition value as false neither true.

Let's say the next problem to clarify I'm trying to describe, say that you have a condition in the transition that tells when you need to walk. Well in that case, you simply say in code, that condition is now true, then you are suppose to be walking. But that condition cannot be used to say in the case is false, that you are not walking, to pass the idle state. You need to create another condition, something like "not_walking" for the previous example.

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

Add a simple check to indicate when transition will activate with the selected condition's value.

Similar to how Unity handles the condition in the transition.

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

Something like this.

//In the transition:
condition conditionName
bool internal_condition_value = //(the check that I previously mention)

// Check if that condition value is the same as the internal condition value
if(conditionName.value == internal_condition_value)
{
// Execute transition.
}

### If this enhancement will not be used often, can it be worked around with a few lines of script?

I think that it should be too dificult to make it, but it's not convenient.

### Is there a reason why this should be core and not an add-on in the asset library?

The reason is simple: Considering that the animation tree is in core of the engine, a simple and yet key function, should the be added to the core engine.

**Comment by Sfoyarbide** [comment]
Sorry my bad, I did not realize that option up until your comment, yet I still trying to find out how to use it.
I search infomation about it in the documentation and the internet, I did not find any information about how to use it. So if you can give some sources on how to use it, I would be totally grateful.

**Comment by Jesusemora** [comment]
> expressions

> it needs to be made more prominent and accessible

no, it is a really good and well done feature and it's perfect the way it is, what it lacks is **documentation**. I had to figure out how to use it myself through experimentation and a lot of research and reading.
 There is barely any mention on how to use expressions evaluation in the documentation or mention of the little `Advance expression Base Node` to select a node to take variables from.
 I would've added what I learned to the documentation, but it's locked behind the github commit system, so its not very encouraging to help with it.

**Sfoyarbide:**

1 - In the animation tree, select a node with script in `Advance Expression Base Node`. Advance->Expression will take ANY variables from this script, even if they are export or not, including things like velocity, which are part of the node itself.

2 - In expression evaluation, type an expression. you can look up what a valid expression is in the docs. but basically, anything you would put after a `if`, such as: `a > 0.5` or `a == b`

And *that's it!* so if you want to evaluate a boolean variable you can just create a variable in your script `var myvalue : bool = false` and do `myvalue` and `not myvalue` in Expression.

**Comment by Sfoyarbide** [comment]
Awesome! Thank you so much for your explanation, It works and I understood!

> I would've added what I learned to the documentation, but it's locked behind the github commit system, so its not very encouraging to help with it.

If there's a way to add this to the documentation, even if it is no encouraging, I think it would be great idea!
If someone knows how to make it happen, please tell us.

**Comment by Calinou** [comment]
> I would've added what I learned to the documentation, but it's locked behind the github commit system, so its not very encouraging to help with it.

You can use GitHub's web-based editor so you don't need to install any software on your PC :slightly_smiling_face:
